src/libmongoc/src/mongoc/mongoc-client-pool.c

```diff
@@ -528,6 +531,15 @@ mongoc_client_pool_set_server_api (mongoc_client_pool_t *pool,
       return false;
    }
 
+   if (pool->client_initialized) {
```

Comment:
Can we use `pool->size > 0` here instead and lose the new flag?  Unless you specified a max pool size of 0 (which I don't think would allow the pool to create any clients at all) then `pool->size` should always be > 0 if a client has been created.

Reply:
On second thought, that would've worked. I had this in previously but refactored this since the logic would allow changes when `pool->size == 0`. Reading the logic in `mongoc_client_pool_push`, this case can't happen, as old clients are only dropped from the pool if `pool->min_pool_size` is truish, in which case you also can't have the last client dropped. I can change it back to the previous check with `pool->size`, but I feel like the flag is more explicit, even though it requires additional memory.